src/DataFrames.jl

```diff
@@ -128,4 +128,18 @@ for (dir, filename) in [
     include(joinpath(dir, filename))
 end
 
+# Import  "StringEncodings" if available and set up decoder
+_decoder=nothing
+ENCODING_PACKAGE_READY = false
+function __init__()
+    try Pkg.installed("StringEncodings")
```

Comment:
For a start, using `Pkg.installed` here is wrong. I'm not sure what do you mean by `StringEncodings` will not be on `LOAD_PATH`.

Also, doing this kind of thing in `__init__` is basically strictly worse than doing it at compile time. If no "optional" dependencies are added or removed, the two methods have the same effect with `__init__` being slower. If optional dependencies are changed, doing it at compile time will require a manual recompilation only when an optional dependency is added, while doing it in `__init__` will break every packages depening on this one, since you suddenly starts to include dependencies that the precompile system did not see before.